test/metabase/lib/js_test.cljs

```diff
@@ -19,37 +19,39 @@
     (is (lib.js/query= q1 q2)))
 
   (testing "explicit fields vs. implied fields"
-    (let [q1 #js {"query" #js {"source-table" 1}}
-          q2 #js {"query" #js {"source-table" 1
-                               "fields" #js [#js ["field" 1 nil]
-                                             #js ["field" 2 nil]
-                                             #js ["field" 3 nil]
-                                             #js ["field" 4 nil]
-                                             #js ["field" 4 nil] ; duplicates are okay
-                                             #js ["field" 4 nil]
-                                             #js ["field" 5 nil]
-                                             #js ["field" 6 nil]
-                                             #js ["field" 7 nil]]}}
-            ;; Note that the order is not relevant; they get grouped.
-            ;; Duplicates are okay, and are tracked.
-          field-ids #js [1 2 6 7 3 5 4 4 4]]
+    (let [q1 #js {"type" "query" "query" #js {"source-table" 1}}
+          q2 #js {"type" "query" "query"
+                  #js {"source-table" 1
+                       "fields" #js [#js ["field" 1 nil]
+                                     #js ["field" 2 nil]
+                                     #js ["field" 3 nil]
+                                     #js ["field" 4 nil]
+                                     #js ["field" 5 nil]
+                                     #js ["field" 6 nil]
+                                     #js ["field" 7 nil]]}}
+          ;; Note that the order is not relevant; they get grouped.
+          ;; Duplicates do NOT make valid queries
```

Comment:
This is a behavior change, we should have a principled reason for choosing one of these.

Per Chesterton's Fence, let's make certain that there's no FE cases broken by this?

More generally, the FE is very sensitive to query equality for whether a question is "dirty" or not, and a bunch of e2e tests will break if it diverges.